mautic/mautic
# Landing Page Draft Feature

MAUT-3816 : Landing Page Draft feature



| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴🟢
| Deprecations?                          | 🔴🟢
| BC breaks? (use the c.x branch)        | 🔴🟢
| Automated tests included?              | 🔴🟢 
| Related user documentation PR URL      | https://github.com/mautic/user-documentation/pull/353
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

This feature is particularly useful if you do not want to immediately publish all your changes and want to get it verified first. Once the changes are approved and ready for sending, the draft can be published. There will be only one draft version of the landing page.

[Original docs from Acquia's Campaign Studio](https://docs.acquia.com/campaign-studio/docs/components/landing-pages/landing-page-basics#section-creating-a-draft-landing-page)

This feature is hidden behind a feature flag. To enable it, add this line to your `config/local.php` file:

```php
page_draft_enabled' => true,
```

You'll see this new button on your landing pages:
![Screenshot 2025-02-27 at 15 23 12](https://github.com/user-attachments/assets/c73e6824-8d38-4e80-8927-1a8573f37d1e)

When editing for the second time you can either apply or discard the draft:
![Screenshot 2025-02-27 at 15 29 24](https://github.com/user-attachments/assets/f872ae7a-cb15-4c1a-8d7f-9bdf4c824bba)







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add ```'page_draft_enabled' => true,``` in your config file.
3. Create a landing page.
4. Go to edit it
5. add some new text in the builder
6. 'Save as draft' button should be visible. Click it.
7. Go to the landing page public URL and see that the text you've added is not there.
8. Go back to edit it. You should see Apply draft and Discard draft buttons. 
9. If you click on Apply Draft, the added text will appear in the public URL of the landing page.
10. If you click on Discard Draft then it will get to the original state even in the edit page.
11. These should function similarly to how they function for emails
